ui/imports/shared/views/chat/ChatTextView.qml

```diff
@@ -0,0 +1,180 @@
+            root.width = 0;
+            chatText.width = Qt.binding(function () {return root.width})
+        } else {
+            chatText.width = Qt.binding(function () {return chatText.implicitWidth})
+            root.width = Qt.binding(function () {return chatText.width})
+        }
+    }
+
+    Component.onCompleted: {
+        root.setWidths()
+    }
+
+    StyledTextEdit {
+        id: chatText
+        visible: !showMoreLoader.active || root.readMore
+        textFormat: Text.RichText
+        wrapMode: Text.Wrap
+        font.pixelSize: Style.current.primaryTextFontSize
+        readOnly: true
+        selectByMouse: true
+        color: Style.current.textColor
+        height: root.veryLongChatText && !root.readMore ? Math.min(implicitHeight, 200) : implicitHeight
+        clip: height < implicitHeight
+        onLinkActivated: {
+
+            root.linkActivated(link)
+            if(link.startsWith("#")) {
+                const channelName = link.substring(1);
+                const foundChannelObj = root.store.chatsModelInst.getChannel(channelName);
+
+                if (!foundChannelObj)
+                {
+                    root.store.chatsModelInst.channelView.joinPublicChat(channelName)
+                    if(root.store.chatsModelInst.communities.activeCommunity.active)
+                    {
+                        root.store.chatsModelInst.channelView.joinPublicChat(channelName)
+                        appMain.changeAppSectionBySectionType(Constants.appSection.chat)
+                    }
+                    return
+                }
+
+                let obj = JSON.parse(foundChannelObj)
+
+                if(obj.chatType === -1 || obj.chatType === Constants.chatTypePublic)
+                {
+                    if(root.store.chatsModelInst.communities.activeCommunity.active) {
+                        root.store.chatsModelInst.channelView.joinPublicChat(channelName)
+                        appMain.changeAppSectionBySectionType(Constants.appSection.chat)
+                    }
+                    root.store.chatsModelInst.channelView.setActiveChannel(channelName);
+                }
+                else if(obj.communityId === root.store.chatsModelInst.communities.activeCommunity.id &&
+                        obj.chatType === Constants.chatTypeCommunity &&
+                        root.store.chatsModelInst.channelView.activeChannel.id !== obj.id
+                        )
+                {
+                    root.store.chatsModelInst.channelView.setActiveChannel(channelName);
+                }
+
+                return
+            }
+
+            if (link.startsWith('//')) {
+                let pk = link.replace("//", "");
+                const userProfileImage = appMain.getProfileImage(pk)
+                openProfilePopup(root.store.chatsModelInst.userNameOrAlias(pk), pk, userProfileImage || root.store.utilsModelInst.generateIdenticon(pk))
+                return;
+            }
+
```

Comment:
can we move some of the logic part to store again in this case? m not sure how much can be separated, but wouldn't it be cleaner that way?

Reply:
done